### PR TITLE
Fixes blob heap size calculation

### PR DESCRIPTION
**Pull request by tmat**
Fixes  https://github.com/dotnet/testimpact/issues/510

### CODE REVIEW

**Comment by tmat** [comment]
@JohnHamby @KevinH-MS

File: src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs [hunk 1]
```diff
@@ -88,12 +88,17 @@ private BlobHandle GetOrAddBlob(ImmutableArray<byte> blob)
                 index = MetadataTokens.BlobHandle(_blobHeapSize);
                 _blobs.Add(blob, index);
 
-                _blobHeapSize += (blob.Length <= 0x7f) ? 1 : ((blob.Length <= 0x3fff) ? 2 : 4) + blob.Length;
```
Comment (JohnHamby) [inline src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs]: I stared at this line for a long time yesterday without figuring out that it had an issue. Reading the correction, I'm still not seeing the nature of the issue.

**Comment by JohnHamby** [comment]
If this fixes the problem, then I sign off enthusiastically and request that it be merged ASAP. We are supposed to get two compiler team members to sign off on any compiler edits, but we're dead in the water until this is fixed.

File: src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs (hunk 1 above)
Comment (tmat) [inline src/Compilers/Core/Portable/PEWriter/MetadataWriter.DynamicAnalysis.cs]: Missing parentheses. ```+``` has precedence over ```?:```. If the blob length was <= 0x7f it added just 1 byte to the size.

**Comment by tmat** [comment]
Gonna merge to unblock folks.

**Comment by KevinH-MS** [comment]
👍

**Pull request merged**